build_tools/cmake/build_and_test_byo_llvm.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Copyright 2020 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Build and test, using CMake/CTest, with the bring-your-own-LLVM path.
+
+set -xeuo pipefail
+
+IREE_BYOLLVM_RELATIVE_BUILD_DIR="${1:-build-byo-llvm}"
+
+source build_tools/cmake/setup_ccache.sh
+
+# Environment variables read by byo_llvm.sh. Absolute paths.
+export IREE_BYOLLVM_BUILD_DIR="$(realpath "${IREE_BYOLLVM_RELATIVE_BUILD_DIR}")"
+export IREE_BYOLLVM_INSTALL_DIR="${IREE_BYOLLVM_BUILD_DIR}/install"
```

Comment:
Fair point. The constraint I'm working under here is that `byo_llvm.sh` is a flexible script that end-users are invited to freely use and combine any part of. It has 3 separate build actions: `build_llvm`, `build_mlir`, `built_iree`.  Like you say, the whole point of "bring your own LLVM", if I understand correctly, is that many users will actually do just that, which means they will skip the `build_llvm` step and start straight at `build_mlir`. But the `build_llvm` action is still there and part of the default suggested usage in the comment at the top of `byo_llvm.sh`, so users will use it, so we should ensure that it works. I can hypothetize that it would be useful in environments where not everyone has a fast build machine, so one person runs `build_llvm`, shares the resulting llvm build, and others can just use that.

We could still do something like symlink the LLVM install path to make it look more like an externally installed LLVM, but I think the current way the build is structured, since byo_llvm.sh defines `IREE_BUILD_BUNDLED_LLVM=OFF`, we effectively treat LLVM as an external install regardless of the path.

Reply:
Yeah, there are definitely tradeoffs. Can you at least add a comment calling out which llvm is being used, and why that is representative enough?

Reply:
Done.